Jenkinsfile

```diff
@@ -15,7 +15,12 @@ node('linux && docker') {
          * to expose the correct versions in headless and atomic.
          */
         stage('Bump build version') {
-          sh 'npx lerna version --conventional-prerelease --amend'
+          sh 'git checkout -- .'
+          sh 'git clean -f'
+          withCredentials([
+          usernameColonPassword(credentialsId: 'github-commit-token', variable: 'GH_CREDENTIALS')]) {
+            sh 'npm run bump:version:pre:amend'
```

Comment:
`bump-version.js` checks out master, which is important for lerna. `lerna version` can't be run in a detached HEAD state.

Reply:
I feel this approach is getting a little complicated, instead of publishing after a bump, could we publish on the next build?

**Original flow**

```
build -> test -> bump -> publish
```

Problem: the build step does not have the correct version.

**Current Solution**

```
Temp bump -> build -> test -> actual bump -> publish
```

But this involves a temp bump, and additional changes in this PR

**Alternatives**

**Option A:**
- If master and untagged: `build -> test -> bump`, and end (this will push the version which will trigger a new build)
- If master and tagged: `build -> test -> publish`

**Option B:**
Bump at the beginning?

```
bump -> build -> test -> publish
```

I prefer the first approach since it separates the version bump and publish steps. Could that simplify the problem?

Reply:
Sounds good, I'll make the tagged commit be published instead.